projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_plugin.py

```diff
@@ -67,6 +70,18 @@ def initialize_job(self, context: JobContext) -> None:
             lambda: _connection_by_configuration(self._impala_cfg),
         )
 
+    @hookimpl(tryfirst=True)
+    def vdk_exception(self, exception: Exception):
+        if is_impala_error(exception):
+            log_and_throw(
```

Comment:
One note - looking at the current `vdk_exception()` wiring, any exception raised within such hook implementation, would not be handled by any plugin in the chain but seems to be raised directly instead. Maybe we need to extend the core hookspec docs on exception raising:
```
 @hookspec
 def vdk_exception(self, exception: Exception) -> bool:
        """
        Called in case the CLI is about to exit with error.

        :return: True if exception is handled and does not need to be logged by main
        """
        pass
```